### PR TITLE
Print warn instead of error logs when there is no container in the ECS task

### PR DESCRIPTION
**Pull request by raymondchen625**
### What does this PR do?
When getting resource tags from an ECS task with zero container, print a warn log instead of error log.

### Motivation
I found intermittent error messages on my Datadog agent (v7.50.3) running as an ECS daemon service like these:

```
timestamp,message
1707489565198,"2024-02-09 14:39:25 UTC | CORE | ERROR | (pkg/workloadmeta/collectors/internal/ecs/ecs.go:252 in getResourceTags) | failed to get client for metadata v3 or v4 API from task ""arn:aws:ecs:us-east-2:123456789012:task/redacted-cluster/***************************bfe75"" and the following containers: []"
1707489568815,"2024-02-09 14:39:28 UTC | PROCESS | ERROR | (pkg/workloadmeta/collectors/internal/ecs/ecs.go:252 in getResourceTags) | failed to get client for metadata v3 or v4 API from task ""arn:aws:ecs:us-east-2:123456789012:task/redacted-cluster/***************************bfe75"" and the following containers: []"
1707489570207,"2024-02-09 14:39:30 UTC | CORE | ERROR | (pkg/workloadmeta/collectors/internal/ecs/ecs.go:252 in getResourceTags) | failed to get client for metadata v3 or v4 API from task ""arn:aws:ecs:us-east-2:123456789012:task/redacted-cluster/***************************bfe75"" and the following containers: []"
```

When the ECS collector polls, it will call the `getResourceTags` function [unless the status is `STOPPED`](https://github.com/raymondchen625/datadog-agent/blob/main/comp/core/workloadmeta/collectors/internal/ecs/ecs.go#L127-L129). According to[ AWS's documentation](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/scheduling_tasks.html#task-lifecycle), there are other statuses like `PROVISIONING`, `ACTIVATING`, and `PENDING`. It might be possible under those statuses, the container array is empty (which can be proven by the logs). the loop never executes and the variable `metaURI` keeps its initial value, an empty string. The error log is [still printed](https://github.com/raymondchen625/datadog-agent/blob/main/comp/core/workloadmeta/collectors/internal/ecs/ecs.go#L271-L274).
This change prints a warn log instead and return when the array is empty.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided. Except if the `qa/skip-qa` label, with required either `qa/done` or `qa/no-code-change` labels, are applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: releasenotes/notes/warn_on_zero_container-fcc1ba6efa55c71d.yaml
Comment (rtrieu) [inline releasenotes/notes/warn_on_zero_container-fcc1ba6efa55c71d.yaml:11]:
```suggestion
    When getting resource tags from an ECS task with zero containers, print a warn log instead of error log.
```

**Review by clamoriniere — APPROVED** [review]
/lgtm

**Comment by clamoriniere** [comment]
hi @rtrieu 
can we get a second review?

**Review by maycmlee — APPROVED** [review]

**Pull request merged**
